app-backend/tile/src/main/scala/routes/MosaicRoutes.scala

```diff
@@ -41,33 +42,34 @@ object MosaicRoutes extends LazyLogging with KamonTraceDirectives {
     pathPrefix("export") {
       optionalHeaderValueByName("Accept") { acceptContentType =>
         traceName("tiles-export-request") {
-          parameter("bbox", "zoom".as[Int]?, "colorCorrect".as[Boolean] ? true) {
-            (bbox, zoom, colorCorrect) => get {
-              complete {
-                val mosaic = Mosaic.render(projectId, zoom, Option(bbox), colorCorrect)
-                val poly = Projected(Extent.fromString(bbox).toPolygon(), 4326)
-                  .reproject(LatLng, WebMercator)(3857)
-                val extent = poly.envelope
-                val future = acceptContentType match {
-                  case Some("image/tiff") => mosaic
-                      .map{ m => MultibandGeoTiff(m, extent, WebMercator) }
+          parameter("bbox", "zoom".as[Int] ?, "colorCorrect".as[Boolean] ? true) {
+            (bbox, zoom, colorCorrect) =>
+              get {
+                complete {
+                  val mosaic = Mosaic.render(projectId, zoom, Option(bbox), colorCorrect)
```

Comment:
It's a route, and it's in a blocking io (!). Is it correct? 
I think that: 

1. all routes should be in a non blocking io (as routes should not block server at all)
2. all blocking operations should be in a blocking io

These two points above would help to make a non blocking server without having blocking routes at all.

What happens here:
`Mosaic.render` produces a blocking future, as the result => `complete { future }` is blocking. 
Probably route easily can be moved into a non blocking pool, by wrapping it into the Future in a non blocking context, so for routes: 

```scala
complete {
  // non blocking pool future
  Future {
    // blocking future // for instance the result of Mosaic operation
  }(non blocking context)
}
```

This idea can be extrapolated on all routes, valid not only for this particular case.

P.S. correct me if I misunderstood smth @echeipesh @notthatbreezy @hectcastro 

Btw, if i described a bit unclear, i created this issue: https://github.com/azavea/raster-foundry/issues/2289
In short: I decided to introduce a separate type to figure out how to split blocking and non blocking ops (Future accepts `ExecutionContext` and Mosaic op accepts `BlockingExecutionContext` in this example):
https://github.com/azavea/raster-foundry/compare/feature/ec-cleanup?expand=1#diff-404f9f22850cae57df7a7ec49db00fd7R48 

That's how blocking Future is wrapped with a non blocking to make route work in async pool and to make operations in a non blocking io.